src/status_im2/contexts/onboarding/events.cljs

```diff
@@ -13,23 +14,28 @@
  (fn [request]
    (status/create-account-and-login request)))
 
-(rf/defn on-delete-profile-success
-  {:events [:onboarding-2/on-delete-profile-success]}
-  [{:keys [db]} key-uid]
-  {:db (update-in db [:multiaccounts/multiaccounts] dissoc key-uid)})
+(re-frame/reg-fx
+ ::validate-mnemonic
+ (fn [[mnemonic on-success on-error]]
+   (status/validate-mnemonic
+    (security/safe-unmask-data mnemonic)
+    (fn [result]
+      (let [{:keys [error]} (types/json->clj result)]
+        (if (seq error)
+          (on-error error)
+          (on-success mnemonic)))))))
+
+(re-frame/reg-fx
+ ::restore-account-and-login
```

Comment:
The beauty of namespaced keywords, but the decision has been made, we should use a different qualification @cammellos, like `multiaccount/restore-and-login` or something along these lines.